fixroute/fixroute.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python3
+"""
+author: Rene Pickhardt (rene.m.pickhardt@ntnu.no)
+Date: 24.1.2020
+License: MIT
+
+This plugin helps you to construct a route object (to be used with sendpay) which goes over a sequence of node ids. if all these nodeids are on a path of payment channels an onion following this path will be constructed. In the case of missing channels `lightning-cli geroute` is invoked to find partial routs. 
+
+This plugin can be used to create circular onions or to send payments along specific paths if that is necessary (as long as the paths provide enough liquidity). 
+
+I guess this plugin could also be used to simulate the behaviour of trampoline payments.
+
+And of course I imagine you the reader of this message will find even more creative ways of using it.
+
+=== Support:
+If you like my work consider a donation at https://patreon.com/renepickhardt or https://tallyco.in/s/lnbook
+"""
+
+from pyln.client import Plugin
+# import lightning
+
+import itertools
+
+
+plugin = Plugin(autopatch=True)
+d = {}
+n = {}
```

Comment:
You can attach these globals to `plugin` instead.

Reply:
yeah I just realized I should also have renamed them as `channelidx` and `nodeidx` I did not set them up in init in the beginning but I thought it would be better to not parse through the json everyone time plugin is called